### PR TITLE
[adapters] Delta: Only skip keys that violate uniqueness.

### PR DESCRIPTION
**Pull request by ryzhyk**
The Delta connector throws an error on unique key violation. When this happens it sbould only skip the affected key, not the entire batch.

## Checklist

- [x] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Documentation updated
- [ ] Changelog updated

## Breaking Changes?

Mark if you think the answer is yes for any of these components:

- [ ] OpenAPI / REST HTTP API / feldera-types / manager ([What is a breaking change?](https://github.com/oasdiff/oasdiff/tree/main))
- [ ] Feldera SQL (Syntax, Semantics)
- [ ] feldera-sqllib (incl. dependencies fxp, etc.) ([What is a breaking change?](https://doc.rust-lang.org/cargo/reference/semver.html#semver-compatibility))
- [ ] Python SDK  ([What is a breaking change?](https://peps.python.org/pep-0387/#backwards-compatibility-rules))
- [ ] fda (CLI arguments)
- [ ] Adapters (including configuration)
- [ ] Storage Format / Checkpoints
- [ ] Others (specify)

### CODE REVIEW

**Review by mihaibudiu — APPROVED** [review]

File: crates/adapters/src/integrated/delta_table/output.rs [hunk 1]
```diff
@@ -696,8 +696,22 @@ async fn stream_encode_and_write(
...
+                    if let Some(controller) = inner.controller.upgrade() {
+                        controller.output_transport_error(
+                            inner.endpoint_id,
+                            &inner.endpoint_name,
+                            false,
+                            e,
+                            Some("delta_uniqueness_violation"),
+                        );
+                    }
+                    continue;
```
Comment (swanandx) [inline crates/adapters/src/integrated/delta_table/output.rs]: we call `continue;` here, means we aren't calling `cursor.step_key()` hence never advancing
```suggestion
                    None
```

> The loop only appears to make progress because of an accidental side effect in non_unique_key_error (util.rs): it does rewind_vals() and walks the values to build its diagnostic — but caps at 10 (counter < 10).


if a single key carries MORE than ten distinct value, we would end up in infinite loop

we should add a test to cover this as well

**Review by swanandx — CHANGES_REQUESTED** [review]
we also need to update stale doc comment on `WriteError::Deterministic` to remove `non-unique keys` as example ( and other places )

**Comment by ryzhyk** [comment]
@swanandx , that's a good catch, thanks! I updated the test to catch the bug.

**Review by swanandx — APPROVED** [review]

File: crates/adapters/src/integrated/delta_table/output.rs [hunk 2]
```diff
@@ -696,8 +696,22 @@ async fn stream_encode_and_write(
         let mut cursor = cursor_builder.build();
 
         while cursor.key_valid() {
-            let op = indexed_operation_type(&inner.value_schema.name, index_name, &mut cursor)
-                .map_err(WriteError::Deterministic)?;
+            let op = match indexed_operation_type(&inner.value_schema.name, index_name, &mut cursor)
```
Comment (mythical-fred) [inline crates/adapters/src/integrated/delta_table/output.rs:699]: Nit: commit body has "sbould" — should be "should".

**Review by mythical-fred — APPROVED** [review]
LGTM — solid test coverage for the skip-on-violation path, including the 11-value key that guards the cursor-advance edge case.
